### PR TITLE
attestation-agent: Extend ResourceUri to support query string

### PR DESCRIPTION
**Pull request by cclaudio**
With this change, the kbs-client is able to send a get-resource request that includes the query string provided e.g.:

kbs-client --url http://127.0.0.1:8080 \
           get-resource \
           --path 'plugin/nebula/credential?ip[ip]=10.11.12.13&ip[netbits]=21&name=pod1'

This is required by the KBS repository plugin interface.

### CODE REVIEW

File: attestation-agent/deps/resource_uri/src/lib.rs [hunk 1]
```diff
@@ -160,6 +167,23 @@ mod tests {
...
             tag: "213".into(),
+            query: None,
+        };
+        assert_eq!(expected, resource);
+    }
+
+    const TEST_QS_URL: &str = "kbs:///plugin/plugname/resourcename?param1=value1&param2=value2";
+
+    #[test]
+    fn deserialize_qs() {
```
Comment (Xynnn007) [inline attestation-agent/deps/resource_uri/src/lib.rs:178]: rstest could help to make the tests more dry. I am sure that you will love it : )

examples https://github.com/confidential-containers/guest-components/blob/main/attestation-agent/attestation-agent/src/eventlog.rs#L158-L174

**Review by Xynnn007 — APPROVED** [review]
LGTM. Thanks @cclaudio for this!

**Review by fitzthum — APPROVED** [review]
LGTM

File: attestation-agent/deps/resource_uri/src/lib.rs (hunk 1 above)
Comment (cclaudio) [reply attestation-agent/deps/resource_uri/src/lib.rs:178]: thanks, rstest definitely makes the code cleaner. Let me rework that test  case

**Comment by cclaudio** [comment]
I will update this PR shortly with a edge case I forgot and also the test case clean-up

**Comment by cclaudio** [comment]
Done.
- patched token_client.rs as well to support query string
- merged all ResourceUri tests into one rstest test with multiple cases. Thanks @Xynnn007  for the feedback.
